uwblueprint/focus-on-nature
# Resolve uncaught error when viewing camp and fix coordinator/counsellor/volunteer editing

## Notion ticket link

[Uncaught error when viewing camp](https://www.notion.so/uwblueprintexecs/Uncaught-error-when-viewing-camp-e7dc9d9ca33e4e72abf171da5fee9869?pvs=4)
[Camp Overview: Fix Camp Edit functionality](https://www.notion.so/uwblueprintexecs/Camp-Overview-Fix-Camp-Edit-functionality-f33023f2ca704b7cb8630fda072a857a?pvs=4)


## Implementation description
* Had to change some of the validators.
* PR #181 refactored create and update validators into a single validator, but turns out we can't have absolutely everything common between the two, so I created two validators (create, update) that implement the old common validator with some minor adjustments.
* After resolving the error, I had to change updateCampById, more information is annotated in the files changed.



## Steps to test
1. Ensure the error is gone when viewing a camp.
2. Test literally everything- all parts of the site including registration, camp creation, camp drafting, camp editing, camp viewing, everything else just to be safe. Notably, make sure that the camp coordinator/counsellor/volunteer component works now. This is to make sure we don't have to redo anything, although the components that I changed were not used anywhere else other than when viewing a camp.



## What should reviewers focus on?
* Correctness, functionality.


## Checklist
